test/images/sample-device-plugin/sampledeviceplugin.go

```diff
@@ -84,14 +91,54 @@ func main() {
 		klog.Errorf("Empty pluginSocksDir")
 		return
 	}
+
+	if autoRegisterString := os.Getenv("AUTO_REGISTER"); autoRegisterString != "" {
+		autoRegister, err = strconv.ParseBool(autoRegisterString)
+		if err != nil {
+			klog.Errorf("invalid boolean value specified")
+			panic(err)
+		}
+
+	}
+
 	socketPath := pluginSocksDir + "/dp." + fmt.Sprintf("%d", time.Now().Unix())
+	triggerPath := pluginSocksDir + "/registered"
 
 	dp1 := plugin.NewDevicePluginStub(devs, socketPath, resourceName, false, false)
 	if err := dp1.Start(); err != nil {
 		panic(err)
 
 	}
 	dp1.SetAllocFunc(stubAllocFunc)
+
+	if !autoRegister {
+		l, err := net.Listen("unix", triggerPath)
+		klog.Infof("Started Listening at %s", triggerPath)
+		if err != nil {
+			panic(err)
+		}
+		defer l.Close()
+
+		sigc := make(chan os.Signal, 1)
+		signal.Notify(sigc, os.Interrupt, syscall.SIGTERM)
+		go func(ln net.Listener, c chan os.Signal) {
+			sig := <-c
+			klog.Infof("Received termination signal %s: shutting down.", sig)
+			ln.Close()
+			os.Exit(0)
+		}(l, sigc)
+
+		klog.InfoS("Starting to accept connections")
+		// Accept new connections
+		conn, err := l.Accept()
```

Comment:
Would it make sense to use simpler synchronisation approach? For example sample device plugin would create a file and wait until it's deleted. e2e test would just delete that file to enable registration?

Reply:
Sure, we can certainly use a file based synchronization approach. I used that in my initial experimentation but pivoted to using a unix socket because I thought it was cleaner. 

I will add additional commits and if you find the file based approach preferable we can get rid of the unix socket completely.

Reply:
It's up to you to decide. I just thought that creating special purpose unix socket in the same directory can confuse people. From other point of view creating file there can confuse them too.

Reply:
I don't feel too strongly either way :) Let's see if @ffromani or @klueska have a preference here.

Reply:
no real preference for me. I _thought_ socket could be simpler but I have zero data to back my claim.